launcher/InstanceList.cpp

```diff
@@ -933,16 +934,19 @@ Task* InstanceList::wrapInstanceTask(InstanceTask* task)
 
 QString InstanceList::getStagedInstancePath()
 {
-    QString key = QUuid::createUuid().toString(QUuid::WithoutBraces);
     QString tempDir = ".LAUNCHER_TEMP/";
+    auto tempPath = FS::PathCombine(m_instDir, tempDir);
+    if (QFileInfo::exists(tempPath)) {
+        FS::deletePath(tempPath);  // clean the path to prevent any collisions
+    }
```

Comment:
Since collisions are so unlikely, maybe keep generating the key until it doesn't exist? Deleting the folder is potentially destructive.
Ideally the temp folder should always be empty though :/. I wonder what causes things to get stuck there.